### ISSUE TITLE
Lost all my config file updates

### ISSUE THREAD

**Issue by tdehaeze** [TheLocehiliosan/yadm#132]
Hello,

Since a few days ago, I try to put all my versionning config files under yadm.
I then worked a lot of hours on my vim config without commiting (I know, this is a bad idea...) and it seems I have lost suddently all my modifications.

When I look at all my yadm commands, my guess is that I have lost my modifications with the following command:
`yadm checkout .dmenurc .`

I am used to use git with magit under emacs so I am not sure is this is normal I have lost all my modifications or not.

Could you help me on that?

Thanks

**Comment by TheLocehiliosan** [comment]
**yadm** works just like **git**. Indeed your `yadm checkout` command is simply translated to a `git checkout` command with the environment configured so your HOME directory is the worktree, and the yadm repo is the GIT_DIR.

`git checkout <paths>` makes the <paths> specified the same as what is currently in the index. If you have added something to the index already, it would revert your files to those versions. However, if you haven't added anything, the index would look like the last commit to your repo. You also specified a `.` as one of the paths. I'm assuming this was in your home directory. The checkout would work recursively on those paths.

Here's an example how how it looks just using git:
```
mkdir example
cd example
git init
mkdir -p a
mkdir -p b
echo 'start a' > a/a.txt
echo 'start b' > b/b.txt
echo 'start c' > c.txt
git add a/a.txt b/b.txt c.txt
git commit -m 'Add initial files'
echo 'changed a' >> a/a.txt
echo 'changed b' >> b/b.txt
echo 'changed c' >> c.txt
cat a/a.txt b/b.txt c.txt
# at this point there are uncommitted changes in all three files
git checkout c.txt .
# that checkout command will restore all the data from the index
cat a/a.txt b/b.txt c.txt
# the files now look like they did after the first commit
```
So I'm doubtful I can help you get your files back. Do you use a terminal multiplexer, that you could see the edits in the history somewhere? That might be a way to get something back.

Can you tell me what you expected the checkout command your ran to do? Then I could help you identify the way to accomplish what you intended.

**Comment by tdehaeze** [comment]
Hi,
Thank you so much for your detailed anwser.
Of course I messed up my git command, it's been one year I am only using git with Magit on Emacs so I forgot some commands...
I just wanted to reset one staged file (.dmenurc).

At the time of the command I was not using tmux. Anyway I just had to redo some work, but it's good now.

I'll try to manage yadm with magit on Emasc using the following config:
```
              (add-to-list 'tramp-methods
                   '("yadm"
                     (tramp-login-program "yadm")
                     (tramp-login-args (("enter")))
                     (tramp-remote-shell "/bin/sh")
                     (tramp-remote-shell-args ("-c"))))
```

I'm closing this issue. Thank you again.
